src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs

```diff
@@ -1070,7 +1072,21 @@ public static OptionSetValue NumberToOptionSet(IRContext irContext, NumberValue[
 
         public static OptionSetValue BooleanToOptionSet(IRContext irContext, BooleanValue[] args)
         {
-            return new OptionSetValue(irContext, "CalculatedOptionSetValue", (OptionSetValueType)irContext.ResultType, (bool)args[0].Value);
+            var optionSetInfo = ((OptionSetValueType)irContext.ResultType)._type.OptionSetInfo;
+
+            if (optionSetInfo != null && optionSetInfo.BackingKind == Core.Types.DKind.Boolean && optionSetInfo.OptionNames.Count() == 2)
+            {
+                foreach (var option in optionSetInfo.OptionNames)
+                {
+                    optionSetInfo.TryGetValue(option, out var value);
+                    if ((bool)value.ExecutionValue == (bool)args[0].Value)

```

Comment:
>value [](http://example.com/codeflow?start=30&length=5)

Any null check here?

Reply:
Good call, checked the result of TryGetValue first.